### PR TITLE
	Add:Rainbow mode,Fix:Window Resize Problem

### LINKED ISSUE #3: Program crashes unexpectedly when resized to be too small
To recreate, begin program, and make window as small as possible. Program will crash. Obviously not a big deal, but thought I'd let you know. Thanks for building this. I use it way more often than I should.

### PR DESCRIPTION
**Pull request by onurtore**
Window resize problem fixed but its a hack.

Fixes #3

### CODE REVIEW

File: cmatrix.c [hunk 1]
```diff
@@ -628,6 +661,33 @@ if (console) {
                         attroff(A_ALTCHARSET);
                     }
                 } else {
+
+                    if(rainbow){
+ 
+                        int randomColor = rand() % 5;
```
Comment (JonnyTech) [inline cmatrix.c]: Should this be 6? IIRC 5 will only generate 0 to 4.

Comment (onurtore) [reply cmatrix.c]: yep you are right

File: cmatrix.c [hunk 2]
```diff
@@ -165,6 +166,9 @@ void *nmalloc(size_t howmuch) {
 void var_init(void) {
     int i, j;
 
+    
+
```
Comment (abishekvashok) [inline cmatrix.c]: Please avoid using 3 empty lines and use only 1

File: cmatrix.c [hunk 3]
```diff
@@ -201,8 +205,9 @@ void var_init(void) {
         spaces[j] = (int) rand() % LINES + 1;
 
         /* And length of the stream */
+    
         length[j] = (int) rand() % (LINES - 3) + 3;
-
+
```
Comment (abishekvashok) [inline cmatrix.c]: There's a tab

File: cmatrix.c [hunk 4]
```diff
@@ -628,6 +661,33 @@ if (console) {
                         attroff(A_ALTCHARSET);
                     }
                 } else {
+
+                    if(rainbow){
+ 
+                        int randomColor = rand() % 6;
+
+                    	switch(randomColor){
```
Comment (abishekvashok) [inline cmatrix.c]: Please properly intend the code

**Review by abishekvashok — CHANGES_REQUESTED** [review]
The code looks good and it works. But it would be best if you could intend it properly.

**Comment by abishekvashok** [comment]
Please check Travis @onurtore

File: cmatrix.spec [hunk 5]
```diff
@@ -0,0 +1,46 @@
+%define name    cmatrix
```
Comment (abishekvashok) [inline cmatrix.spec]: This file is dynamically generated from .spec.in file using auto tools for distribution as an rpm file. Please don't include it.

**Review by abishekvashok — CHANGES_REQUESTED** [review]

**Review by abishekvashok — APPROVED** [review]
Thanks

**Pull request merged**